R/comparisons.R

```diff
@@ -0,0 +1,239 @@
+  stopifnot(
+    "Object should be of class comparison" =
+      (is.comparison(object)),
+    "'comparison' object does not contain the correct attributes" =
+      (all(
+        c(
+          "data", "baseline"
+        ) %in% attributes(object)$names
+      )
+      ),
+    "Comparison must be a list of 'scenario' objects" =
+      (is.list(object$data) && (
+        all(
+          vapply(
+            object$data, is.scenario,
+            FUN.VALUE = TRUE
+          )
+        )
+      )),
+    "Baseline must be among scenario names" =
+      (is.character(object$baseline) &&
+        object$baseline %in% names(object$data))
+  )
+  invisible(object)
+}
+
+#' @export
+print.comparison <- function(x, ...) {
+  # prepare information
+  header <- cli::style_bold("Scenario comparison object")
+  scenario_count <- glue::glue(" # Scenarios: {length(x$data)}")
+  data_status <- ifelse(
+    !all(
+      vapply(x$data, sce_has_data, FUN.VALUE = TRUE)
+    ),
+    cli::col_magenta(
+      " Some scenarios have no data, use `run_scenario()` to prepare data"
+    ),
+    " All scenario data are prepared, use `sce_get_outcomes()` to get data"
+  )
+  baseline <- glue::glue(" Baseline scenario: {cli::col_blue(x$baseline)}")
+  # the scenario matching variables
+  matching_variables <- c(
+    " Scenario matching variables:",
+    ifelse(is.na(x$match_variables),
+      cli::col_magenta("  No matching variables specified!"),
+      cli::col_green(glue::glue("  {x$match_variables}"))
+    )
+  )
+  # the output comparison variables
+  comparison_variables <- c(
+    " Scenario comparison variables:",
+    ifelse(is.na(x$comparison_variables),
+      cli::col_magenta("  No comparison variables specified!"),
+      cli::col_green(glue::glue("  {x$comparison_variables}"))
+    )
+  )
+  # the model function(s)
+  model_fun <- c(
+    " Model functions found:",
+    cli::col_cyan(
+      glue::glue("  {unique(unlist(lapply(x$data, `[`, 'model_function')))}")
+    )
+  )
+  # output all
+  writeLines(
+    c(
+      header, scenario_count, baseline, data_status,
+      matching_variables, comparison_variables, model_fun
+    )
+  )
+  invisible(x)
+}
+
+#' Check whether an object is a 'comparison'
+#'
+#' @param x An R object.
+#'
+#' @return A logical indicating whether the object inherits from the class
+#' 'comparison'.
+#'
+#' @export
+#'
+#' @examples
+#' # prepare two scenarios of the final size of an epidemic
+#' pandemic_flu <- scenario(
+#'   model_function = "finalsize::final_size",
+#'   parameters = make_parameters_finalsize_UK(r0 = 1.5),
+#'   replicates = 1L
+#' )
+#'
+#' covid19 <- scenario(
+#'   model_function = "finalsize::final_size",
+#'   parameters = make_parameters_finalsize_UK(r0 = 5.0),
+#'   replicates = 1L
+#' )
+#'
+#' # create a comparison object
+#' x <- comparison(
+#'   pandemic_flu = pandemic_flu, covid19 = covid19,
+#'   baseline = "pandemic_flu"
+#' )
+#'
+#' is.comparison(x)
+is.comparison <- function(x) {
```

Comment:
`is_comparison()` for consistent naming (especially given S3 dispatch is used in the package this function name may be confusing)

Reply:
Fixed in https://github.com/epiverse-trace/scenarios/pull/12/commits/4401b480777a73ad1c65b137dadcf42ce00a3650